### DISCUSSION TITLE
Custom Trace ID Generator in Open Telemetry .NET

### DISCUSSION THREAD

**Discussion by kiro1993** [open-telemetry/opentelemetry-dotnet#5084] [General]
Hi,

I'm looking for a way to customize generation of open telemetry trace id. This is part of how I'm initializing the tracer in the Startup.cs.

services.AddOpenTelemetryTracing(tracingBuilder =>
    tracingBuilder
        .AddSource(TracingConstants.ServiceName)
        .AddAspNetCoreInstrumentation());
        
        
I tried using something like .SetIdGenerator(customIDGenerator); but couldn't do it.

these are the nuget versions im using:
  "OpenTelemetry" Version="[1.2.0-rc3]" 
  "OpenTelemetry.Api" Version="[1.2.0-rc3]"

**Comment by cijothomas** [comment]
> I tried using something like .SetIdGenerator(customIDGenerator); but couldn't do it.

Can you share the exact code you tried so we can check what was missing?

**Reply by kiro1993** [reply]
@cijothomas Please find the open telemetry versions used in the main question.       
I tried using it in multiple ways:
1. 
![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/13682023/73a3b838-953b-4980-8e41-498cc6e7d07a)
    
2.
![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/13682023/9504a96c-2111-4ffc-980c-ac9b67f943f2)

**Comment by WeihanLi** [comment]
Would `Activity.TraceIdGenerator` help?

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.traceidgenerator

**Reply by cijothomas** [reply]
^ is the right API to achieve custom traceid generator. Thank you!
